lib/matplotlib/legend.py

```diff
@@ -507,10 +512,16 @@ def __init__(self, parent, handles, labels,
         self._loc_used_default = tmp  # ignore changes done by _set_loc
 
         # figure out title fontsize:
-        if title_fontsize is None:
-            title_fontsize = mpl.rcParams['legend.title_fontsize']
-        tprop = FontProperties(size=title_fontsize)
-        self.set_title(title, prop=tprop)
+        self.title_prop = FontProperties._from_any(title_prop)
```

Comment:
Do we need to hold onto this and expose it as public API?  Previously we did not hold onto it and it does not look like that the user can update this after the fact and have it be respected.